samples/algorithms/MajoranaQubits/src/GateSet.qs

```diff
@@ -0,0 +1,82 @@
+/// A set of gates built upon the custom measurements
+/// provided by the hardware provider.
+///
+/// Source:
+///  [1] Surface code compilation via edge-disjoint paths
+///      https://arxiv.org/pdf/2110.11493
+
+/// Apply a CNOT gate to the given qubits.
+/// Source: [1] Figure 3.
+operation CNOT(control : Qubit, target : Qubit) : Unit {
+    // Prepare an ancilla qubit in the |+⟩ state.
+    use ancilla = Qubit();
+    PrepareX(ancilla);
+
+    let a = Mzz(control, ancilla);
+    let b = Mxx(ancilla, target);
+    let c = Mz(ancilla);
+    Reset(ancilla);
+
+    if b == One {
+        Z(control);
+    }
+
+    if a != c {
+        X(target);
+    }
+}
+
+
+/// Prepare a qubit in the |+⟩ state.
+/// `q` must be in the |0⟩ state.
```

Comment:
This isn't true anymore, actually. Because this uses Mx, it is guaranteed to project to the X basis regardless of where the qubit started.